src/libclipper/include/clipper/task_executor.hpp

```diff
@@ -272,9 +291,19 @@ class TaskExecutor {
             VersionedModelId vm = VersionedModelId(
                 container_info["model_name"], container_info["model_version"]);
             int replica_id = std::stoi(container_info["model_replica_id"]);
-            active_containers_->add_container(
-                vm, std::stoi(container_info["zmq_connection_id"]), replica_id,
-                parse_input_type(container_info["input_type"]));
+              active_containers_->add_container(
+                      vm, std::stoi(container_info["zmq_connection_id"]), replica_id,
+                      parse_input_type(container_info["input_type"]));
+
+            auto model_info = redis::get_model(redis_connection_, vm);
+            VersionedModelId model_id = VersionedModelId(model_info["model_name"], model_info["model_version"]);
```

Comment:
The value of this variable will be identical to `vm` (defined on line 291) and can be deleted